server/src/main/java/org/elasticsearch/search/retriever/StandardRetrieverBuilder.java

```diff
@@ -0,0 +1,162 @@
+            return terminateAfter;
+        }, TERMINATE_AFTER_FIELD, ObjectParser.ValueType.INT);
+
+        PARSER.declareField((r, v) -> r.sortBuilders = v, (p, c) -> {
+            List<SortBuilder<?>> sortBuilders = SortBuilder.fromXContent(p);
+            c.trackSectionUsage(NAME + ":" + SORT_FIELD.getPreferredName());
+            return sortBuilders;
+        }, SORT_FIELD, ObjectParser.ValueType.OBJECT_ARRAY);
+
+        PARSER.declareField((r, v) -> r.minScore = v, (p, c) -> {
+            float minScore = p.floatValue();
+            c.trackSectionUsage(NAME + ":" + MIN_SCORE_FIELD.getPreferredName());
+            return minScore;
+        }, MIN_SCORE_FIELD, ObjectParser.ValueType.FLOAT);
+
+        PARSER.declareField((r, v) -> r.collapseBuilder = v, (p, c) -> {
+            CollapseBuilder collapseBuilder = CollapseBuilder.fromXContent(p);
+            if (collapseBuilder.getField() != null) {
+                c.trackSectionUsage(COLLAPSE_FIELD.getPreferredName());
+            }
+            return collapseBuilder;
+        }, COLLAPSE_FIELD, ObjectParser.ValueType.OBJECT);
+
+        RetrieverBuilder.declareBaseParserFields(NAME, PARSER);
+    }
+
+    public static StandardRetrieverBuilder fromXContent(XContentParser parser, RetrieverParserContext context) throws IOException {
+        if (context.clusterSupportsFeature(NODE_FEATURE) == false) {
+            throw new ParsingException(parser.getTokenLocation(), "unknown retriever [" + NAME + "]");
+        }
+        return PARSER.apply(parser, context);
+    }
+
+    private QueryBuilder queryBuilder;
+    private SearchAfterBuilder searchAfterBuilder;
+    private int terminateAfter = SearchContext.DEFAULT_TERMINATE_AFTER;
+    private List<SortBuilder<?>> sortBuilders;
+    private Float minScore;
+    private CollapseBuilder collapseBuilder;
+
+    @Override
+    public void extractToSearchSourceBuilder(SearchSourceBuilder searchSourceBuilder) {
+        if (preFilterQueryBuilders.isEmpty() == false) {
+            BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+
+            for (QueryBuilder preFilterQueryBuilder : preFilterQueryBuilders) {
+                boolQueryBuilder.filter(preFilterQueryBuilder);
+            }
+
+            if (queryBuilder != null) {
+                boolQueryBuilder.must(queryBuilder);
+            }
+
+            searchSourceBuilder.subSearches().add(new SubSearchSourceBuilder(boolQueryBuilder));
+        } else if (queryBuilder != null) {
+            searchSourceBuilder.subSearches().add(new SubSearchSourceBuilder(queryBuilder));
+        }
+
+        if (searchSourceBuilder.searchAfter() == null) {
+            if (searchAfterBuilder != null) {
+                searchSourceBuilder.searchAfter(searchAfterBuilder.getSortValues());
+            }
+        } else {
+            throw new IllegalArgumentException("[search_after] cannot be declared on multiple retrievers");
```

Comment:
To reflect this error message,  think we should structure this and following `if` statemenst differently:

```java
 if (searchAfterBuilder != null) {
    if (searchSourceBuilder.searchAfter() == null) {
        throw new IllegalArgumentException("[sort] cannot be declared on multiple retrievers"); 
    } else {
        searchSourceBuilder.sort(sortBuilders);
    }
}
```

In this case if there are 2 standard retrievers, but only one of them declares `min_score` we allow it. But if 2 standard retrievers declare min_score, we don't allow that. WDYT?

Reply:
I added a parameter to extraction called `compound`. If a compound retriever is a direct or indirect parent of a retriever, it can make decisions based on that. In this case, I went ahead and disallowed anything except for `query` as part of the `standard` retriever for now if a its a child of a compound retriever. This gives us the maximum flexibility moving forward to reduce bwc issues as we add addtiional per-query fields.